features/device_key/source/DeviceKey.cpp

```diff
@@ -253,7 +253,7 @@ int DeviceKey::generate_key_by_trng(uint32_t *output, size_t& size)
 
     final_size = 0;
     in_size = size;
-    while (true) {
+    while (DEVICEKEY_SUCCESS == ret && final_size < size) {
```

Comment:
I think the `DEVICEKEY_SUCCESS` check can be omitted here.

Reply:
@hanno-arm 
I agree.